lib/blockchain_double.js

```diff
@@ -1152,6 +1127,16 @@ BlockchainDouble.prototype.getTransaction = function(hash, callback) {
 
 BlockchainDouble.prototype.getTransactionReceipt = function(hash, callback) {
   hash = to.hex(hash);
+  const pendingTxs = this.pending_transactions;
+
+  if (pendingTxs.length > 0) {
```

Comment:
No need for the `length > 0` check here, we can let the `for` loop logic handle the 0 length case.